server/image_remove.go

```diff
@@ -17,41 +16,36 @@ func (s *Server) RemoveImage(ctx context.Context, req *pb.RemoveImageRequest) (r
 		recordOperation(operation, time.Now())
 		recordError(operation, err)
 	}()
+	resp = &pb.RemoveImageResponse{}
 
-	logrus.Debugf("RemoveImageRequest: %+v", req)
+	logrus.Debugf("%s: request: %+v", operation, req)
 	image := ""
 	img := req.GetImage()
 	if img != nil {
 		image = img.Image
 	}
 	if image == "" {
-		return nil, fmt.Errorf("no image specified")
+		logrus.Infof("%s: no image specified in request", operation)
+		return resp, nil
 	}
-	var (
-		images  []string
-		deleted bool
-	)
-	images, err = s.StorageImageServer().ResolveNames(s.systemContext, image)
+
+	images, err := s.StorageImageServer().ResolveNames(s.systemContext, image)
 	if err != nil {
 		if err == storage.ErrCannotParseImageID {
 			images = append(images, image)
 		} else {
-			return nil, err
+			logrus.Infof("%s: unable to resolve image %q: %v", operation, image, err)
+			return resp, nil
```

Comment:
I think we need to return this error. Otherwise, we'd mask errors from, for instance, `sysregistriesv2`.